scipy/sparse/_base.py

```diff
@@ -600,6 +599,8 @@ def _mul_dispatch(self, other):
         if issparse(other):
             if self.shape[1] != other.shape[0]:
                 raise ValueError('dimension mismatch')
+            if other.ndim == 1:
+                raise ValueError('Cannot yet multiply a 1d sparse array')
```

Comment:
I might prefer to do `_mul_sparse_matrix(other.reshape(other._shape_as_2d))` here. What do you think?